RefRed/calculations/lr_data.py

```diff
@@ -96,19 +86,19 @@ def __init__(self, workspace, lconfig=None, is_data=True, parent=None):
         self.attenuatorNbr = mt_run.getProperty('vATT').value[0] - 1
         self.date = mt_run.getProperty('run_start').value
 
-        sample = self.workspace.getInstrument().getSample()
-        source = self.workspace.getInstrument().getSource()
+        sample = workspace.getInstrument().getSample()
+        source = workspace.getInstrument().getSource()
         self.dMS = sample.getDistance(source)
 
         # create array of distances pixel->sample
-        self.number_x_pixels = int(self.workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])  # 256
-        self.number_y_pixels = int(self.workspace.getInstrument().getNumberParameter("number-of-y-pixels")[0])
+        self.number_x_pixels = int(workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])  # 256
+        self.number_y_pixels = int(workspace.getInstrument().getNumberParameter("number-of-y-pixels")[0])
 
         dPS_array = np.zeros((self.number_x_pixels, self.number_y_pixels))
```

Comment:
Can you get rid of this big array? It's only used in line 105 and only one single element is used, out of the whole array